admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,391 @@
+                        "reason": "KernelOops",
+                        "pattern": "BUG: unable to handle kernel NULL pointer dereference at .*"
+                },
+                {
+                        "type": "temporary",
+                        "reason": "KernelOops",
+                        "pattern": "divide error: 0000 \\[#\\d+\\] SMP"
+                },
+                {
+                        "type": "permanent",
+                        "condition": "KernelDeadlock",
+                        "reason": "AUFSUmountHung",
+                        "pattern": "task umount\\.aufs:\\w+ blocked for more than \\w+ seconds\\."
+                },
+                {
+                        "type": "permanent",
+                        "condition": "KernelDeadlock",
+                        "reason": "DockerHung",
+                        "pattern": "task docker:\\w+ blocked for more than \\w+ seconds\\."
+                }
+        ]
+    }
+----
+
+<1> Rules and conditions that apply to Docker images.
+<2> Monitoring services, in a comma-separated list.
+<3> Path to the monitoring service log.
+<4> List of events to be monitored.
+<5> Label to indicate the error is an event (`temporary`) or NodeCondition (`permanent`).
+<6> Text message to describe the error.
+<7> Error message that the Node Problem Detector watches for. 
+<8> Rules and conditions that apply to the kernel.
+
+////
+https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#node-problem-detector
+The Node Problem Detector supports file-based kernel logging. However, it is easy to extend it to support other log formats.
+////
+
+To configure the Node Problem Detector, add or remove problem conditions and events.
+
+. Edit the Node Problem Detector configuration map with a text editor.
++
+[source,bash]
+----
+oc edit configmap -n openshift-node-problem-detector node-problem-detector
+----
+
+. Remove, add, or edit any NodeConditions or events as needed. 
++
+[source,yaml]
+----
+{
+       "type": <`temporary` or `permanent`>,
+       "reason": <free-form text describing the error>,
+       "pattern": <log message to watch for>
+},
+----
++
+For example:
++
+[source,yaml]
+----
+{
+       "type": "temporary",
+       "reason": "UnregisterNetDevice",
+       "pattern": "unregister_netdevice: waiting for \\w+ to become free. Usage count = \\d+"
+},
+----
+
+. Restart running pods to apply the changes. To restart pods, you can delete all existing pods:
++
+[source,bash]
+----
+# oc delete pods -n openshift-node-problem-detector -l name=node-problem-detector
+----
+
+. To display Node Problem Detector output to standard output (stdout) and standard error (stderr) 
+add the following to the configuration map:
++
+[source,yaml]
+----
+spec:
+  template:
+    spec:
+      containers:
+      - name: node-problem-detector
+        command:
+        - node-problem-detector
+        - --alsologtostderr=true <1>
+          --log_dir="/tmp" <2>
```

Comment:
This line needs a dash in the same column as the rows above and below it. The `command` parameter expects a list of strings, and each item in the list in YAML is designated by a single dash before it. Sorry I didn't notice before.